File: port-scanner.cpp
#include <iostream>
#include <sys/socket.h>
#include <arpa/inet.h>
#include <netinet/in.h>
#include <unistd.h>
#include <cstring>




int main(){


    //variables as neeeded
    
    const char* host_ip = "127.0.0.1"; //change IP

    struct sockaddr_in socketaddr;
    
    
    

    for(int ports = 0; ports <= 65535; ++ports){
        int sock = socket(AF_INET, SOCK_STREAM, 0);    
        memset(&socketaddr, 0, sizeof(socketaddr));  // zero out space in memory
        socketaddr.sin_family = AF_INET;
        //socketaddr.sin_addr.s_addr = inet_addr(INADDR_ANY);
        socketaddr.sin_port = htons(ports);
        inet_pton(AF_INET, host_ip, &socketaddr.sin_addr);

        
        if (connect(sock, (struct sockaddr*)&socketaddr, sizeof(socketaddr)) == 0){
            std::cout << "\e[38;5;212mport: " << ports << " is open!" << std::endl;
        }
        close(sock);
    }

  
    



    return 0;
}

File: port-scanner.py
#!/usr/bin/python3
import socket




def scan():  
    host = "127.0.0.1"   # change IP
    for ports in range(0, 65535):
        
        try:
            s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
            s.settimeout(0.5)
            if s.connect_ex((host, ports)) == 0:
                print(f"\033[35mPort {ports} is open!\033[0m")

            else:
                continue
        #error handling    
        except socket.gaierror as e:
            print(f"\033[31mInvalid Hostname: {e}")
            break
        
        except socket.error as e:
            print(f"\033[31mInvalid Socket: {e}")
            break
            
              
        
        
if __name__ == "__main__":
    scan()
